Projects/Application/Sources/Common/Network/Http/Http.swift

```diff
@@ -0,0 +1,51 @@
+//
+// Created by Maxim Berezhnoy on 2019-02-12.
+// Licensed under the MIT license
+//
+//
+
+import Foundation
+
+// MARK: - RequestAdapter
+protocol RequestAdapter {
+    func adapt(urlRequest: URLRequest) -> URLRequest
+}
+
+// MARK: - RequestRetrier
+protocol RequestRetrier {
+    func should(retry request: Http.Request, with response: Http.Response, retryAfter: (TimeInterval) -> Void)
+}
+
+// MARK: - Http core
+final class Http {
+    enum Method {
+        case get
+        case put
+        case post
+        case delete
+    }
+
+    typealias Headers = [String: String]
+
+    var adapter: RequestAdapter?
+    var retrier: RequestRetrier?
+
+    private let urlSession: URLSession
+
+    static let `default` = {
+       Http(configuration: .default)
+    }()
+
+    init(configuration: Configuration) {
+        let sessionConfiguration = URLSessionConfiguration.default
+        sessionConfiguration.timeoutIntervalForRequest = configuration.requestTimeout
+        sessionConfiguration.httpAdditionalHeaders =
+                Configuration.defaultHeaders.merging(configuration.additionalHeaders) { _, r in r }
+
+        self.urlSession = URLSession(configuration: sessionConfiguration)
+    }
+
+    func request(_ url: URL, _ method: Method = .get, body: Data? = nil) -> Request {
```

Comment:
I would create an additional initializer in `Http.Request` for that purpose

Reply:
It's not clear from the implementation, but the `request` function is actually meant to create and start the request; user shouldn't be able to manually create requests.

Reply:
I want to think about naming; maybe it can be improved somehow. Right now there is a confusion about the request; it is a type that holds all the information about the request but also performs a job over it.

Reply:
Probably the biggest reason I decided on that naming was my inspiration from Alamofire; it has the same convention, with `Request` being a task and holding a reference to its state, `URLRequest`.
But I agree with you, it makes things less clear and writing `request.request` to access the `URLRequest` is not ideal.

I think the name `RequestTask` suits well.

Reply:
Maybe even `HTTP.Task` or `HTTP.Job`

Reply:
Yes, I think a name like `Http.Task` provides enough information.

Reply:
Fixed